src/js/overlay.js

```diff
@@ -311,6 +312,10 @@ Overlay.prototype.realign = function(dimension, size) {
 		if (!this.opts.arrow) {
 			this.wrapper.style['margin' + utils.capitalise(edge)] = -(this.wrapper['offset' + utils.capitalise(dimension)]/2) + 'px';
 		}
+		// Set via JS as it needs the full size when calculating the negative margins
+		// and setting left: 50% for example before, if the overlay doesn't fit, would
+		// shrink the overlay
```

Comment:
This comment can be improved to read a bit better. I get what it means because we discussed it yesterday but I am not sure I'd understand if we hadn't.